sunbeam-python/sunbeam/jobs/common.py

```diff
@@ -389,3 +395,36 @@ def str_presenter(dumper: yaml.Dumper, data: str) -> yaml.ScalarNode:
     if data.count("\n") > 0:
         return dumper.represent_scalar("tag:yaml.org,2002:str", data, style="|")
     return dumper.represent_scalar("tag:yaml.org,2002:str", data)
+
+
+def _get_default_no_proxy_settings() -> set:
+    """Return default no proxy settings"""
+    return {
+        "127.0.0.1",
+        "localhost",
+        K8S_CLUSTER_SERVICE_CIDR,
+        K8S_CLUSTER_POD_CIDR,
+        ".svc",
+    }
+
+
+def get_proxy_settings(deployment: Deployment) -> dict:
+    proxy = {}
+    client = deployment.get_client()
+    try:
+        proxy_from_db = read_config(client, PROXY_CONFIG_KEY) or {}
+        proxy = {
+            p.upper(): v
+            for p in ("http_proxy", "https_proxy", "no_proxy")
+            if (v := proxy_from_db.get("proxy", {}).get(p))
+        }
```

Comment:
That's a bit hard to read:
```suggestion
        proxy_from_db = read_config(client, PROXY_CONFIG_KEY).get("proxy")
        proxy = {
            p.upper(): v
            for p in ("http_proxy", "https_proxy", "no_proxy")
            if (v := proxy_from_db.get(p))
        }
```

Reply:
Done